src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,189 @@ void VulnerabilityScannerFacade::start(
+                            catch (const std::exception& e)
+                            {
+                                logWarn(WM_VULNSCAN_LOGTAG,
+                                        "Discarded dbsync_packages event: %s package. %s",
+                                        (delta->data_as_dbsync_packages() && delta->data_as_dbsync_packages()->name())
+                                            ? delta->data_as_dbsync_packages()->name()->c_str()
+                                            : "",
+                                        e.what());
+                            }
+                        }
+                    }
+                }
+            });
+
+        m_syncThread = std::thread(
+            [scanOrchestrator, this]()
+            {
+                while (!m_stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (m_stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    while (!m_syncQueue->empty() && !m_stopRetryThread.load())
+                    {
+                        std::shared_ptr<std::string> returnStr = m_syncQueue->pop();
+
+                        flatbuffers::Verifier syncVerifier(reinterpret_cast<const uint8_t*>(returnStr->c_str()),
+                                                           returnStr->size());
+
+                        if (SyscollectorSynchronization::VerifySyncMsgBuffer(syncVerifier))
+                        {
+                            const SyscollectorSynchronization::SyncMsg* syncMsg =
+                                SyscollectorSynchronization::GetSyncMsg(
+                                    reinterpret_cast<const char*>(returnStr->c_str()));
+                            std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*>
+                                data = syncMsg;
+
+                            try
+                            {
+                                logDebug1(WM_VULNSCAN_LOGTAG,
+                                          "Retry syscollector_packages event: %s package",
+                                          (syncMsg->data_as_state() &&
+                                           syncMsg->data_as_state()->attributes_as_syscollector_packages() &&
+                                           syncMsg->data_as_state()->attributes_as_syscollector_packages()->name())
+                                              ? syncMsg->data_as_state()
+                                                    ->attributes_as_syscollector_packages()
+                                                    ->name()
+                                                    ->c_str()
+                                              : "");
```

Comment:
This expression is really hard to follow, can we declare it as a const variable before the debug call with a descriptive name?

Reply:
ditto